pyproject.toml

```diff
@@ -90,3 +90,12 @@ dependencies = [
 [tool.setuptools]
 # This is actually not a python project, we just use pyproject.toml to manage dependencies.
 py-modules = []
+
+[tool.ruff]
+target-version = "py311"
+line-length = 100
+
+[tool.ruff.lint]
+preview = true
```

Comment:
I think we shouldn't depend on unstable features

Reply:
As far as I'm aware it's not "features" but rules. The `explicit-preview-rules = true` means you don't get any preview rules unless you explicitly enable them.

This is necessary to enable `W391` for parity with the existing CI flake8 rules.